nominatim/tokenizer/sanitizers/config.py

```diff
@@ -102,30 +106,41 @@ def get_delimiter(self, default: str = ',;') -> Pattern[str]:
         return re.compile('\\s*[{}]+\\s*'.format(''.join('\\' + d for d in delimiter_set)))
 
 
-    def get_filter_kind(self, *default: str) -> Callable[[str], bool]:
-        """ Return a filter function for the name kind from the 'filter-kind'
-            config parameter.
+    def get_filter(self, param: str, default: Optional[Sequence[str]] = None
+            ) -> Callable[[Optional[str]], bool]:
+        """ Returns a filter function for the given parameter of the sanitizer
+            configuration.
 
-            If the 'filter-kind' parameter is empty, the filter lets all items
-            pass. If the parameter is a string, it is interpreted as a single
-            regular expression that must match the full kind string.
-            If the parameter is a list then
-            any of the regular expressions in the list must match to pass.
+            The value provided for the parameter in sanitizer configuration
+            should be a string or list of strings, where each string is a regular
+            expression. These regular expressions will later be used by the
+            filter function to filter strings.
 
             Arguments:
-                default: Filters to be used, when the 'filter-kind' parameter
-                         is not specified. If omitted then the default is to
-                         let all names pass.
+                param: The parameter for which the filter function
+                       will be created.
+                default: Takes a list of strings where each string
+                         is a regular expression.These regular expressions will
+                         be used by the filter function in case the given parameter
+                         is missing in the sanitizer configuration.
+                         Note that if this default is not provided, the filter function
+                         lets all target strings pass, including None type.
 
             Returns:
-                A filter function which takes a name string and returns
-                True when the item passes the filter.
+                A filter function that takes a target string as the argument and
+                returns True if the it fully matches any of the regular expressions
+                otherwise returns False.
+                Note that if instead of target string, None type is provided then
+                it simply returns False.
         """
-        filters = self.get_string_list('filter-kind', default)
+        values = self.data.get(param, None)
+        filters = self.get_string_list(param) if values is not None else default
 
-        if not filters:
+        if filters is None:
```

Comment:
This changes the behaviour when the parameter is present but empty. It's an odd edge case but the idea was to have it behave as if the parameter was not present at all.

Reply:
I see. This is less intuitive unless explicitly mentioned. What I can think of is - we can do either of two things-
1. Go with these changes and let `[]` keep its meaning i.e allow nothing. Because to allow anything to pass we can just skip the parameter, which ideally makes sense.
2. We can revert back to the original code and explicitly mention that `[]` is just like skipping the parameter. Also with that, if I allow `get_string_list` to take `None` default and return an empty list for it, we could bring back the `get_string_list(param, default)` thing.

Please let me know what you think and then I will give it a go.